docs/source/graphbolt/ondisk-dataset.rst

```diff
@@ -0,0 +1,829 @@
+              - name: negative_dsts
+                format: numpy
+                in_memory: true
+                path: set_lp/val_negative_dsts.npy
+        test_set:
+          - data:
+              - name: node_pairs
+                format: numpy
+                in_memory: true
+                path: set_lp/test_node_pairs.npy
+              - name: negative_dsts
+                format: numpy
+                in_memory: true
+                path: set_lp/test_negative_dsts.npy
+
+
+For the graph structure, number of nodes is specified by the ``num`` field and
+edges are stored in a csv file in format of ``<src, dst>`` like below.
+
+.. code:: csv
+
+    edges.csv
+
+    0,1
+    1,2
+    2,3
+    3,4
+    4,5
+    5,6
+    6,7
+    7,8
+    8,9
+
+
+For the feature data, we have feature data named as ``feat`` for nodes and
+edges. The feature data are stored in numpy files in shape of ``[num_nodes, 10]``
+and ``[num_edges, 10]`` respectively like below.
+
+.. code:: python
+
+    node_feat.npy
+
+    [[0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
+     [1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
+     [2. 2. 2. 2. 2. 2. 2. 2. 2. 2.]
+     [3. 3. 3. 3. 3. 3. 3. 3. 3. 3.]
+     [4. 4. 4. 4. 4. 4. 4. 4. 4. 4.]
+     [5. 5. 5. 5. 5. 5. 5. 5. 5. 5.]
+     [6. 6. 6. 6. 6. 6. 6. 6. 6. 6.]
+     [7. 7. 7. 7. 7. 7. 7. 7. 7. 7.]
+     [8. 8. 8. 8. 8. 8. 8. 8. 8. 8.]
+     [9. 9. 9. 9. 9. 9. 9. 9. 9. 9.]]
+
+    edge_feat.npy
+
+    [[0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
+     [1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
+     [2. 2. 2. 2. 2. 2. 2. 2. 2. 2.]
+     [3. 3. 3. 3. 3. 3. 3. 3. 3. 3.]
+     [4. 4. 4. 4. 4. 4. 4. 4. 4. 4.]
+     [5. 5. 5. 5. 5. 5. 5. 5. 5. 5.]
+     [6. 6. 6. 6. 6. 6. 6. 6. 6. 6.]
+     [7. 7. 7. 7. 7. 7. 7. 7. 7. 7.]
+     [8. 8. 8. 8. 8. 8. 8. 8. 8. 8.]
+     [9. 9. 9. 9. 9. 9. 9. 9. 9. 9.]]
+
+For the ``tasks`` field, we have two tasks: ``node_classification`` and
+``link_prediction``. For each task, we have three sets: ``train_set``,
+``validation_set`` and ``test_set``.
+
+For ``node_classification`` task, we have two fields: ``seed_nodes`` and
+``labels``. The ``seed_nodes`` field is used to specify the node IDs for
+training and evaluation. The ``labels`` field is used to specify the
+labels. Both of them are stored in numpy files with shape of ``[num_nodes]``
+like below.
+
+.. code:: python
+
+    train_seed_nodes.npy
+
+    [0 1 2 3 4 5]
+
+    train_labels.npy
+
+    [0 1 0 1 0 1]
+
+    val_seed_nodes.npy
+
+    [6 7]
+
+    val_labels.npy
+
+    [0 1]
+
+    test_seed_nodes.npy
+
+    [8 9]
+
+    test_labels.npy
+
+    [0 1]
+
+
+For ``link_prediction`` task, we have two fields: ``node_pairs``,
+``negative_dsts``.  The ``node_pairs`` field is used to specify the node pairs.
+The ``negative_dsts`` field is used to specify the negative destination nodes.
+They are stored in numpy file with shape of ``[num_edges, 2]`` and
+``[num_edges, num_neg_dsts]`` respectively like below.
+
+.. code:: python
+
+    train_node_pairs.npy
+
+    [[0 1]
```

Comment:
Is it the common way to show npy data with space instead of , to separate items?

Reply:
I represent like this just because it's for illustration only. I don't know whether it's a common way.

Reply:
better to keep it consistent with npy data.

Reply:
ok

Reply:
align with print np.array